test/unit/LibPrices.js

```diff
@@ -455,4 +482,237 @@ describe("Unit tests: LibPrices.sol", function () {
+                    )
+
+                    let funding = await applyFundingWrapper(
+                        userPosition,
+                        globalCumulativeRate,
+                        userCumulativeRate
+                    )
+
+                    expect(funding[0]).to.equal(expectedUserPosition[0]) // check quote is changed
+                    expect(funding[1]).to.equal(expectedUserPosition[1]) // check base is unchanged
+                })
+            }
+        )
+
+        context(
+            "When the global rate is less than the user rate & user has a short position",
+            async () => {
+                it("applies a negative change to the user's position", async () => {
+                    quote = ethers.utils.parseEther("0")
+                    base = ethers.utils.parseEther("-50")
+
+                    userPosition = [quote, base]
+
+                    let globalCumulativeRate = ethers.utils.parseEther("2200")
+                    let userCumulativeRate = ethers.utils.parseEther("2300")
+
+                    // Funding applied (i.e. quote diff) will be (-50 base * (-100 quote / base) = 5000 quote
+                    // Since user had zero quote, they should have -5000 now (take negative of funding applied)
+                    let expectedUserPosition = calculateExpectedUserPosition(
+                        quote,
+                        base,
+                        globalCumulativeRate,
+                        userCumulativeRate
+                    )
+                    expect(expectedUserPosition[0]).to.equal(
+                        ethers.utils.parseEther("-5000")
+                    )
+
+                    let funding = await applyFundingWrapper(
+                        userPosition,
+                        globalCumulativeRate,
+                        userCumulativeRate
+                    )
+
+                    expect(funding[0]).to.equal(expectedUserPosition[0]) // check quote is changed
+                    expect(funding[1]).to.equal(expectedUserPosition[1]) // check base is unchanged
+                })
+            }
+        )
+    })
+
+    describe("applyInsurance", async () => {
+        context("when insurance funding has increased", async () => {
+            it("returns the correct insurance/user positions", async () => {
+                let userPosition = [
+                    ethers.utils.parseEther("0"),
+                    ethers.utils.parseEther("110"),
+                ] // quote, base
+                let insurancePosition = [
+                    ethers.utils.parseEther("0"),
+                    ethers.utils.parseEther("0"),
+                ] // quote, base
+                let insuranceGlobalRate = [0, ethers.utils.parseEther("10"), 0] // timestamp, fundingRate, cumulativeFundingRate
+                let insuranceUserRate = [0, ethers.utils.parseEther("5"), 0] // timestamp, fundingRate, cumulativeFundingRate
```

Comment:
@kumar-ish, @BenjaminPatch, @raymogg . The applyInsurance function does calculations based on the fundingRate not the cumulativeFundingRate. I think this is a major error as user's would be have differing insurance payments based on when they settle their accounts, it should be cumulativeFundingRate instead. Is this correct?

Reply:
@OsmanBran Yes, that's correct; fix accordingly